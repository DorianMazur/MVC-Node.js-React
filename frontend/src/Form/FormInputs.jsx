import React from 'react';
import { Form, Segment } from 'semantic-ui-react';
import PropTypes from 'prop-types';

class FormInputs extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
    this.handleChange = this.handleChange.bind(this);
  }

  handleChange(event) {
    this.props.updateData({ [event.target.id]: event.target.value });
  }

  render() {
    const { firstName, lastName, email } = this.props;
    return (
      <Segment>
        <Form.Input
          id="firstName"
          fluid
          icon="user"
          iconPosition="left"
          placeholder="First Name"
          value={firstName}
          onChange={this.handleChange}
        />
        <Form.Input
          id="lastName"
          fluid
          icon="user outline"
          iconPosition="left"
          placeholder="Last Name"
          value={lastName}
          onChange={this.handleChange}
        />
        <Form.Input
          id="email"
          fluid
          icon="mail"
          iconPosition="left"
          placeholder="Email"
          value={email}
          onChange={this.handleChange}
        />
      </Segment>
    );
  }
}

FormInputs.propTypes = {
  firstName: PropTypes.string,
  lastName: PropTypes.string,
  email: PropTypes.string,
  updateData: PropTypes.func,
};

export default FormInputs;
