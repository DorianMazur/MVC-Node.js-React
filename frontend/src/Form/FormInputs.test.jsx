import '@testing-library/jest-dom';
import React from 'react';

import { render } from '@testing-library/react';
import FormInputs from './FormInputs.jsx';

const props = {
  firstName: 'Test',
  lastName: 'Test',
  email: 'Test@Test.pl',
  updateData: jest.fn(),
};

test('FormInputs render correctly', () => {
  const container = render(<FormInputs {...props} />);
  expect(container.baseElement).toMatchSnapshot();
});
