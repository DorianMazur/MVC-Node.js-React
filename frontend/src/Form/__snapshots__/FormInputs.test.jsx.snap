// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInputs render correctly 1`] = `
<body>
  <div>
    <div
      class="ui segment"
    >
      <div
        class="field"
      >
        <div
          class="ui fluid left icon input"
        >
          <input
            id="firstName"
            placeholder="First Name"
            type="text"
            value="Test"
          />
          <i
            aria-hidden="true"
            class="user icon"
          />
        </div>
      </div>
      <div
        class="field"
      >
        <div
          class="ui fluid left icon input"
        >
          <input
            id="lastName"
            placeholder="Last Name"
            type="text"
            value="Test"
          />
          <i
            aria-hidden="true"
            class="user outline icon"
          />
        </div>
      </div>
      <div
        class="field"
      >
        <div
          class="ui fluid left icon input"
        >
          <input
            id="email"
            placeholder="Email"
            type="text"
            value="Test@Test.pl"
          />
          <i
            aria-hidden="true"
            class="mail icon"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;
